scripts/publish.ts

```diff
@@ -10,16 +10,39 @@ import { logging, tags } from '@angular-devkit/core';
 import { spawnSync } from 'child_process';
 import * as semver from 'semver';
 import { packages } from '../lib/packages';
+import { wombat } from '../lib/registries';
 import build from './build';
 
-
 export interface PublishArgs {
   tag?: string;
```

Comment:
make tag non-optional?

Reply:
Semantically `--tag` is required, but when given the `PublishArgs` type, we haven't validated that yet. We can't assume that any arguments are provided in this type.

This is a really unfortunate system, so there's not a great way of doing this. I think the best solution would be to make `PublishArgs` semantically correct (`tag` is required), but then in the default exported function accept `Partial<PublishArgs>`, that way everything is forced to be possibly `undefined` and must be manually validated at runtime.

Even that wouldn't really handle incorrectly typed values so it's not quite fool-proof (such as the `--branchCheck=false` case), but the only way to handle that would be to accept an `unknown` or a `Record<string, unknown>`... I can't think of a better solution that doesn't involve re-writing the way we process commands as a whole. WDYT?